src/js/expansionPanel.directive.js

```diff
@@ -102,15 +102,17 @@ function expansionPanelDirective() {
     vm.onRemove = onRemove;
     vm.init = init;
 
-    $attrs.$observe('disabled', function(disabled) {
-      isDisabled = (typeof disabled === 'string' && disabled !== 'false') ? true : false;
+	isDisabled = (typeof $attrs.disabled === 'string' && $attrs.disabled !== 'false') ? true : false;
 
-      if (isDisabled === true) {
-        $element.attr('tabindex', '-1');
-      } else {
-        $element.attr('tabindex', '0');
-      }
-    });
+	setTabIndex();
+
+	$scope.$watch($attrs.ngDisabled, function(disabled) {
```

Comment:
You could shorten the entire thing to just this.
Remove you initial ngDisabled check
Remove your setTabIndex Function

```javascript
if ($attrs.ngDisabled !== undefined) {
  scope.$watch($attrs.ngDisabled, function(value) {
    isDiabled = value;
    $element.attr('tabindex', value ? -1 : 0);
  });
}
```